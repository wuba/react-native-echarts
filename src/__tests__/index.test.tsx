/**
 * @jest-environment @shopify/react-native-skia/jestEnv.mjs
 */
// tests for basic props
import React from 'react';
import { render } from '@testing-library/react-native';
import SkiaComponent from '../skiaChart';
import SVGComponent from '../svgChart';
const Components = [SkiaComponent, SVGComponent];
const svg = `<svg width="100" height="100" viewBox="0 0 100 100">`;
const node = {
  tag: 'svg',
  attrs: {
    width: 100,
    height: 100,
    viewBox: '0 0 100 100',
  },
};

Components.forEach((Component) => {
  describe(`${Component.displayName} Component` || 'unknown', () => {
    it('renders correctly', () => {
      const { toJSON } = render(<Component />);
      expect(toJSON()).toMatchSnapshot();
    });

    it('renders Canvas when svgString is provided', () => {
      const { queryByTestId, toJSON } = render(
        <Component svg={svg} node={node} />
      );
      expect(toJSON()).toMatchSnapshot();
      expect(queryByTestId('component')).not.toBeNull();
    });

    it('does not render Canvas when svgString is not provided', () => {
      const { queryByTestId, toJSON } = render(<Component />);
      expect(toJSON()).toMatchSnapshot();
      expect(queryByTestId('component')).toBeNull();
    });

    it('renders GestureHandler when handleGesture is true and useRNGH is ture', () => {
      const { queryByTestId, toJSON } = render(
        <Component svg={svg} node={node} useRNGH />
      );
      expect(toJSON()).toMatchSnapshot();
      expect(queryByTestId('gesture-handler')).not.toBeNull();
      expect(queryByTestId('pan-responder-handler')).toBeNull();
    });

    it('renders PanResponderHandler when handleGesture is true and useRNGH is false', () => {
      const { queryByTestId, toJSON } = render(
        <Component svg={svg} node={node} useRNGH={false} />
      );
      expect(toJSON()).toMatchSnapshot();
      expect(queryByTestId('gesture-handler')).toBeNull();
      expect(queryByTestId('pan-responder-handler')).not.toBeNull();
    });

    it('does not render GestureHandler when handleGesture is false', () => {
      const { queryByTestId, toJSON } = render(
        <Component svg={svg} node={node} handleGesture={false} />
      );
      expect(toJSON()).toMatchSnapshot();
      expect(queryByTestId('gesture-handler')).toBeNull();
      expect(queryByTestId('pan-responder-handler')).toBeNull();
    });
  });
});
