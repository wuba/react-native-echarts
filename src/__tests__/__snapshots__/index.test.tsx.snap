// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkiaChart Component does not render Canvas when svgString is not provided 1`] = `null`;

exports[`SkiaChart Component does not render GestureHandler when handleGesture is false 1`] = `
<View
  style={
    {
      "height": 0,
      "width": 0,
    }
  }
  testID="component"
/>
`;

exports[`SkiaChart Component renders Canvas when svgString is provided 1`] = `
<View
  style={
    {
      "height": 0,
      "width": 0,
    }
  }
  testID="component"
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      {
        "backgroundColor": "rgba(0,0,0,0)",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      }
    }
    testID="pan-responder-handler"
  />
</View>
`;

exports[`SkiaChart Component renders GestureHandler when handleGesture is true and useRNGH is ture 1`] = `
<View
  style={
    {
      "height": 0,
      "width": 0,
    }
  }
  testID="component"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgba(0,0,0,0)",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      }
    }
    testID="gesture-handler"
  />
</View>
`;

exports[`SkiaChart Component renders PanResponderHandler when handleGesture is true and useRNGH is false 1`] = `
<View
  style={
    {
      "height": 0,
      "width": 0,
    }
  }
  testID="component"
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      {
        "backgroundColor": "rgba(0,0,0,0)",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      }
    }
    testID="pan-responder-handler"
  />
</View>
`;

exports[`SkiaChart Component renders correctly 1`] = `null`;

exports[`SvgChart Component does not render Canvas when svgString is not provided 1`] = `null`;

exports[`SvgChart Component does not render GestureHandler when handleGesture is false 1`] = `
<View
  style={
    {
      "height": 100,
      "width": 100,
    }
  }
  testID="component"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="100"
    bbWidth="100"
    focusable={false}
    height={100}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 100,
          "width": 100,
        },
      ]
    }
    vbHeight={100}
    vbWidth={100}
    width={100}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    />
  </RNSVGSvgView>
</View>
`;

exports[`SvgChart Component renders Canvas when svgString is provided 1`] = `
<View
  style={
    {
      "height": 100,
      "width": 100,
    }
  }
  testID="component"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="100"
    bbWidth="100"
    focusable={false}
    height={100}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 100,
          "width": 100,
        },
      ]
    }
    vbHeight={100}
    vbWidth={100}
    width={100}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    />
  </RNSVGSvgView>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      {
        "backgroundColor": "rgba(0,0,0,0)",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      }
    }
    testID="pan-responder-handler"
  />
</View>
`;

exports[`SvgChart Component renders GestureHandler when handleGesture is true and useRNGH is ture 1`] = `
<View
  style={
    {
      "height": 100,
      "width": 100,
    }
  }
  testID="component"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="100"
    bbWidth="100"
    focusable={false}
    height={100}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 100,
          "width": 100,
        },
      ]
    }
    vbHeight={100}
    vbWidth={100}
    width={100}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    />
  </RNSVGSvgView>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgba(0,0,0,0)",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      }
    }
    testID="gesture-handler"
  />
</View>
`;

exports[`SvgChart Component renders PanResponderHandler when handleGesture is true and useRNGH is false 1`] = `
<View
  style={
    {
      "height": 100,
      "width": 100,
    }
  }
  testID="component"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="100"
    bbWidth="100"
    focusable={false}
    height={100}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 100,
          "width": 100,
        },
      ]
    }
    vbHeight={100}
    vbWidth={100}
    width={100}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    />
  </RNSVGSvgView>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      {
        "backgroundColor": "rgba(0,0,0,0)",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      }
    }
    testID="pan-responder-handler"
  />
</View>
`;

exports[`SvgChart Component renders correctly 1`] = `null`;
